### PR TITLE
Deprecate configmapprovider package, replace with mapconverter

### PR DESCRIPTION
**Pull request by bogdandrutu**
The new `config/mapconverter` has the same structure as the `config/mapprovider` so it is consistent.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: config/mapconverter/expandmapconverter/expand.go [hunk 1]
```diff
@@ -21,10 +21,10 @@ import (
 	"go.opentelemetry.io/collector/config"
 )
 
-// NewExpandConverter returns a service.ConfigMapConverterFunc, that expands all environment variables for a given config.Map.
+// New returns a config.MapConverterFunc, that expands all environment variables for a given config.Map.
 //
 // Notice: This API is experimental.
```
Comment (codeboten) [inline config/mapconverter/expandmapconverter/expand.go:26]: Is this notice still valid after this change?

File: config/configmapprovider/deprecated.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// limitations under the License.
+
+package configmapprovider // import "go.opentelemetry.io/collector/config/configmapprovider"
+
+import (
+	"go.opentelemetry.io/collector/config/mapconverter/expandmapconverter"
+	"go.opentelemetry.io/collector/config/mapconverter/overwritepropertiesmapconverter"
+)
+
+// Deprecated: use expandmapconverter.New.
```
Comment (codeboten) [inline config/configmapprovider/deprecated.go:22]:
```suggestion
// Deprecated: [v0.49.0] use expandmapconverter.New.
```

File: config/configmapprovider/deprecated.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+var NewExpandConverter = expandmapconverter.New
+
+// Deprecated: use overwritepropertiesmapconverter.New.
```
Comment (codeboten) [inline config/configmapprovider/deprecated.go:25]:
```suggestion
// Deprecated: [v0.49.0] use overwritepropertiesmapconverter.New.
```

File: config/mapconverter/overwritepropertiesmapconverter/properties_test.go [hunk 4]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package configmapprovider
+package overwritepropertiesmapconverter
```
Comment (codeboten) [inline config/mapconverter/overwritepropertiesmapconverter/properties_test.go:15]: not sure about package name, seems a bit wordy... I'll admit that I don't have a better alternative for name though

Comment (dmitryax) [reply config/mapconverter/overwritepropertiesmapconverter/properties_test.go:15]: But it'll inconsistent with `expandmapconterter.New`, so I'd probably chose to keep `overwritepropertiesmapconverter` unless we can change `expandmapconterter` as well

File: config/mapconverter/expandmapconverter/expand.go (hunk 1 above)
Comment (bogdandrutu) [reply config/mapconverter/expandmapconverter/expand.go:26]: Let's keep changes separately. Not sure that only because we moved the location does this change to not be experimental.

File: config/mapconverter/overwritepropertiesmapconverter/properties_test.go (hunk 4 above)
Comment (bogdandrutu) [reply config/mapconverter/overwritepropertiesmapconverter/properties_test.go:15]: I would keep it as is, since we have a consistent way of naming.

File: config/configmapprovider/deprecated.go (hunk 2 above)
Comment (bogdandrutu) [reply config/configmapprovider/deprecated.go]: Done.

File: config/configmapprovider/deprecated.go (hunk 3 above)
Comment (bogdandrutu) [reply config/configmapprovider/deprecated.go]: Done.

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
